### PR TITLE
Add Ansible Rook role

### PR DESCRIPTION
**Pull request by erzetpe**

### CODE REVIEW

File: ansible/playbooks/roles/rook/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+rook_helm_chart_file_name: rook-ceph-v1.8.1.tgz
+rook_chart_namespace: rook-ceph
+rook_helm_chart_name: rook-ceph
+rook_enabled: true
```
Comment (atsikham) [inline ansible/playbooks/roles/rook/defaults/main.yml]: Should this var be in defaults with no ability to overwrite for user?

File: ansible/playbooks/roles/rook/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+kubeconfig:
```
Comment (atsikham) [inline ansible/playbooks/roles/rook/defaults/main.yml]: this is available as a group var for all hosts, not necessary here

File: ansible/playbooks/roles/rook/defaults/main.yml (hunk 1 above)
Comment (atsikham) [reply ansible/playbooks/roles/rook/defaults/main.yml]: The idea from @to-bar is to enable rook via feature-mapping

File: ansible/playbooks/roles/rook/defaults/main.yml (hunk 2 above)
Comment (erzetpe) [reply ansible/playbooks/roles/rook/defaults/main.yml]: Done.

**Comment by atsikham** [comment]
Would be nice to have a documentation with basic usage examples.

**Comment by erzetpe** [comment]
It's still a draft. :)

File: ansible/playbooks/roles/rook/defaults/main.yml (hunk 1 above)
Comment (erzetpe) [reply ansible/playbooks/roles/rook/defaults/main.yml]: This was temporary.

File: docs/changelogs/CHANGELOG-2.0.md [hunk 3]
```diff
@@ -21,6 +21,8 @@
 - [#2945](https://github.com/epiphany-platform/epiphany/issues/2945) - epicli apply sleeps 10 seconds after creating inventory
 - [#2940](https://github.com/epiphany-platform/epiphany/issues/2940) - firewalld.service unit could not be found on host however ansible_facts sees it as defined
 
+- [#2669](https://github.com/epiphany-platform/epiphany/issues/2669) - Restarting the installation process can cause certificate problems if K8s was not fully configured
```
Comment (atsikham) [inline docs/changelogs/CHANGELOG-2.0.md]: whyyy 🤔

File: ansible/playbooks/rook.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+---
+- hosts: all
+  gather_facts: true
+  tasks: []
+
+- hosts: kubernetes_master
+  become: true
+  become_method: sudo
+  roles:
+    - rook
```
Comment (atsikham) [inline ansible/playbooks/rook.yml:10]: kubeconfig env here

File: docs/changelogs/CHANGELOG-2.0.md (hunk 3 above)
Comment (erzetpe) [reply docs/changelogs/CHANGELOG-2.0.md]: Will be fixed with rebase later on. Still a draft. :)

File: ansible/playbooks/rook.yml (hunk 4 above)
Comment (erzetpe) [reply ansible/playbooks/rook.yml:10]: This will be similar to approach in the other roles. Legit.

Comment (erzetpe) [reply ansible/playbooks/rook.yml:10]: Done.

File: schema/common/defaults/configuration/rook.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+kind: configuration/rook
```
Comment (atsikham) [inline schema/common/defaults/configuration/rook.yml:2]: How about starting with ---?

Comment (erzetpe) [reply schema/common/defaults/configuration/rook.yml:2]: I was thinking about this.

Comment (erzetpe) [reply schema/common/defaults/configuration/rook.yml:2]: Done.

File: ansible/playbooks/rook.yml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+---
+- hosts: all
```
Comment (to-bar) [inline ansible/playbooks/rook.yml]: Do we need to gather facts from all hosts (including e.g. `postgresql` and `kafka`)?

The next play will gather facts second time for `kubernetes_master`.

We do not use caching facts yet, there is issue #810 to implement.

Comment (erzetpe) [reply ansible/playbooks/rook.yml]: We don't need. Probably only repository is needed there.

Comment (erzetpe) [reply ansible/playbooks/rook.yml]: Done. :)

**Pull request merged**